### PR TITLE
Add a readme for running docker image

### PR DESCRIPTION
**Pull request by nirinchev**

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    some_empty_folder="$(mktemp -d)" # mount an empty folder to /apps. If not done, the docker image will import a sample app which will mess up the test setup
+    baas_version="2022-05-16" # the version of the image you'd like to use. See versions at https://github.com/realm/ci/pkgs/container/ci%2Fmongodb-realm-test-server
+
+    docker run \
+      -e MONGODB_REALM_HOSTNAME=$baas_hostname -e AWS_ACCESS_KEY_ID=$baas_access_key -e AWS_SECRET_ACCESS_KEY=$baas_secret_key \
+      --mount type=bind,src=$some_empty_folder,dst=/apps \
+      -u 1000:1000 \
+      -p 9090:9090 -p 26000:26000 \ # Expose the baas and mongodb ports to the host. 9090 is baas and 26000 is mongodb.
+      -it \ # interactive and attach tty
+      --rm \ # clean up the container at exit. omit this if you want to preserve the state between runs. resume with https://docs.docker.com/engine/reference/commandline/start/
```
Comment (desistefanova) [inline test/README.md:62]:
```suggestion
      --rm \ # clean up the container at exit. omit this if you want to preserve the state between runs resume with https://docs.docker.com/engine/reference/commandline/start/
```
```suggestion
      --rm \ # clean up the container at exit. omit this if you want to preserve the state between runs. resume with https://docs.docker.com/engine/reference/commandline/start/
```

Comment (desistefanova) [inline test/README.md:62]:
```suggestion
      --rm \ # clean up the container at exit. omit this if you want to preserve the state between runs resume with https://docs.docker.com/engine/reference/commandline/start/
```
```suggestion
      --rm \ # clean up the container at exit. omit this if you want to preserve the state between runs. resume with https://docs.docker.com/engine/reference/commandline/start/
```

Comment (desistefanova) [inline test/README.md:62]:
```suggestion
      --rm \ # clean up the container at exit. omit this if you want to preserve the state between runs resume with https://docs.docker.com/engine/reference/commandline/start/
```

Comment (desistefanova) [inline test/README.md:62]:
```suggestion
      --rm \ # clean up the container at exit. Omit this if you want to preserve the state between runs. Resume with https://docs.docker.com/engine/reference/commandline/start/
```

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Running the tests
```
Comment (desistefanova) [inline test/README.md:1]: This README.md file is internal, right? It is not exposed in the API doc.

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Run integration tests against Atlas cluster on MongoDB cloud.
+
+In order to run the tests that requires a backend, you need to setup a MongoDB cluster to run against.
+A free-tier database should be enough for most contributors.
+
+1) Log on to [MongoDB cloud](cloud.mongodb.com)
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply test/README.md:8]: dartdoc requires full path.

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+```
+BAAS_URL=https://realm.mongodb.com
+BAAS_CLUSTER=<cluster_name> # probably Cluster0
+BAAS_API_KEY=<public_key>
+BAAS_PRIVATE_API_KEY=<private_key>
+BAAS_PROJECT_ID=<project_id>
+```
+10) Now you can run `dart test` and it should include the integration tests.
+
+If you are a MongoDB employee, you can instead choose to run the tests against [cloud-dev](cloud-dev.mongodb.com).
```
Comment (desistefanova) [inline test/README.md:32]:
```suggestion
If you are a MongoDB employee, you can instead choose to run the tests against [cloud-dev](https://cloud-dev.mongodb.com).
```

File: test/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+2. Request API keys for BaaS - reach out to Tim Sedgwick or Mike O'Brien for help.
+3. [Configure Docker for use with GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry) - you only need the authentication part.
+4. Take note of your IP address (it has to be the actual IP address, not localhost or 127.0.0.1).
+5. Run the docker image:
+    ```sh
+    baas_hostname="10.0.1.123"# $ the IP address from step   4.
+    baas_access_key="<public_key>" # the credentials we got in step 2.
+    baas_secret_key="<private_key>" # the credentials we got in step 2.
+    some_empty_folder="$(mktemp -d)" # mount an empty folder to /apps. If not done, the docker image will import a sample app which will mess up the test setup
+    baas_version="2022-05-16" # the version of the image you'd like to use. See versions at https://github.com/realm/ci/pkgs/container/ci%2Fmongodb-realm-test-server
```
Comment (nielsenko) [inline test/README.md:54]: The url doesn't work

File: test/README.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+    docker run \
+      -e MONGODB_REALM_HOSTNAME=$baas_hostname -e AWS_ACCESS_KEY_ID=$baas_access_key -e AWS_SECRET_ACCESS_KEY=$baas_secret_key \
+      --mount type=bind,src=$some_empty_folder,dst=/apps \
+      -u 1000:1000 \
+      -p 9090:9090 -p 26000:26000 \ # Expose the baas and mongodb ports to the host. 9090 is baas and 26000 is mongodb.
```
Comment (nielsenko) [inline test/README.md:60]: Putting comments behind \ prevents copy paste

File: test/README.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      -it \ # interactive and attach tty
+      --rm \ # clean up the container at exit. Omit this if you want to preserve the state between runs. Resume with https://docs.docker.com/engine/reference/commandline/start/
+      ghcr.io/realm/ci/mongodb-realm-test-server:$baas_version
+    ```
```
Comment (nielsenko) [inline test/README.md:64]: I'm stuck here. I get:
Unable to find image 'ghcr.io/realm/ci/mongodb-realm-test-server:latest' locally
docker: Error response from daemon: Head "https://ghcr.io/v2/realm/ci/mongodb-realm-test-server/manifests/latest": unauthorized.
See 'docker run --help'.

**Review by nielsenko — CHANGES_REQUESTED** [review]

File: test/README.md (hunk 7 above)
Comment (desistefanova) [reply test/README.md:64]: Me too. You have to do docker login first. But I was not able to login. Will test this later again.

Comment (nielsenko) [reply test/README.md:64]: Missed a point.

File: test/README.md (hunk 6 above)
Comment (blagoev) [reply test/README.md:60]: also does not work on Windows probably

**Comment by blagoev** [comment]
I don't know why but my comments are not added behind existing comments but are separated.

File: test/README.md (hunk 6 above)
Comment (desistefanova) [reply test/README.md]: @blagoev I added powershell commands for windows.

File: test/README.md (hunk 5 above)
Comment (desistefanova) [reply test/README.md]: Done

File: test/README.md (hunk 7 above)
Comment (desistefanova) [reply test/README.md:116]: Try now with the new instruction and I hope you will be able to authenticate for authorized access.

**Review by desistefanova — APPROVED** [review]

**Review by desistefanova — APPROVED** [review]
Commands are tested and worked.

**Review by blagoev — APPROVED** [review]

File: test/README.md (hunk 6 above)
Comment (blagoev) [reply test/README.md]: Not ideal. We don't require powershell anywhere else

File: test/README.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+    * Don't forget to Copy the new token.
+
+    * Then run the following commands:
+      - For MacOS anf Linux:
+      ```sh
+        $ export CR_PAT=YOUR_TOKEN
+        $ echo $CR_PAT | docker login ghcr.io -u USERNAME --password-stdin
+      ```
+      - For Windows powershell:
```
Comment (blagoev) [inline test/README.md:61]: lets not require powershell here

**Review by blagoev — CHANGES_REQUESTED** [review]

**Review by blagoev — APPROVED** [review]

**Pull request merged**